### ISSUE TITLE
Labels for task>0

### ISSUE THREAD

**Issue by Areeb2735** [zhoudw-zdw/CIL_Survey#4]
Hi.

Thank you for this great Library for Continual Learning.

I wanted to ask regarding the label mapping when we are at the task 2 of class incremental training. Let's say I am training a simple finetuning model. 

I am training on CIFAR100 on an increment of 10 classes in each task. The first training will have labels [0,1,2,3,4,5,6,7,8,9]. For the second task, the labels are [10,11,12,13,14,15,16,17,18,19]. Do you map these to [0,1,2,3,4,5,6,7,8,9] or use some other loss function instead of cross-entropy? If you map it, can you please point out the code piece. And how do you handle this at inference time?

Thanks in Advance.

**Comment by zhoudw-zdw** [comment]
Hi @Areeb2735, thanks for your interest in our work.

The classes are not mapped to [0,1,2,3,4,5,6,7,8,9], and they are actually in the format of [10,11,12,13,14,15,16,17,18,19]. You can refer to this function for more details. https://github.com/zhoudw-zdw/CIL_Survey/blob/791728c072974da264d985d7e9c4825fdc3d17ff/utils/data_manager.py#L137C5-L137C5

Additionally, you can print the labels during training to see the labels.

**Comment by Areeb2735** [comment]
Thanks. 

But then how is the loss computed? Given that you are using cross-entropy loss, the targets are the index. Now the targets are [10,11,12,13,14,15,16,17,18,19].

**Comment by zhoudw-zdw** [comment]
In class-incremental learning, the classifier is expanded after each task.

So you have a classifier that outputs 20 dims at the second stage, 30 for the third stage, etc.

See here for the classifier update.

 https://github.com/zhoudw-zdw/CIL_Survey/blob/791728c072974da264d985d7e9c4825fdc3d17ff/models/finetune.py#L43

**Comment by Areeb2735** [comment]
Thank you
